### PR TITLE
Add NetFlow OCSF mapping package

### PR DESCRIPTION
**Pull request by mavam**
## 🔍 Problem

`read_netflow` produces typed NetFlow v5, NetFlow v9, and IPFIX records, but the Library has no reusable path for normalizing them to OCSF.

## 🛠️ Solution

- Add the `netflow` package and `netflow::ocsf::map`.
- Map flow records to OCSF 1.9.0 Network Activity.
- Represent an unknown connection initiator with `initiator_id: 0`.
- Keep flow observation bounds and duration within `traffic`.
- Preserve options records and unmapped source fields.
- Cover real NetFlow and IPFIX captures plus synthetic mapping edge cases.

## 💬 Review

Please focus on the OCSF semantics for initiators, counters, timestamps, NAT, and options records.

All ten tests pass against the local engine's `1.9.0-dev` schema. With the shipping `metadata.version: 1.9.0`, Tenzir v6.9.0-rc.1 drops events in `ocsf::derive` because it does not bundle the stable schema yet. Keep this PR blocked until stable OCSF 1.9.0 support lands. Do not ship development metadata.

<sub>
✅ Closes TNZ-870<br>
📎 Related: tenzir/mono#40
</sub>

### CODE REVIEW

File: netflow/operators/ocsf/events/network_activity.tql [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+$event.ocsf.is_src_dst_assignment_known = false
+
+// --- OCSF: occurrence attributes --------------
+
+if $event.source.flow_start? != null {
+  $event.ocsf.start_time = move $event.source.flow_start
+} else if $event.source.flow_start_seconds? != null {
+  $event.ocsf.start_time = move $event.source.flow_start_seconds
+} else if $event.source.flow_start_milliseconds? != null {
+  $event.ocsf.start_time = move $event.source.flow_start_milliseconds
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql]: Fixed in 68cc8e2. The mapper now retains `flow_start`/`flow_end` as the first choice, then prefers nanoseconds, microseconds, milliseconds, and seconds in that order. The regression test populates every precision.

File: netflow/operators/ocsf/events/network_activity.tql [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      $event.ocsf.dst_endpoint.proxy_endpoint.ip = move $event.source.post_nat_destination_ipv6_address
+    }
+  }
+}
+
+$event._protocol = $event.source.protocol_identifier?
+$event.ocsf.connection_info = {
+  direction_id: 0,
+  protocol_num: move $event.source.protocol_identifier?,
+}
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql]: I don't think this field occurs within the mapper's `read_netflow` input contract. Community ID is not an [IANA IPFIX information element](https://www.iana.org/assignments/ipfix/ipfix.xhtml)—the registry's community fields are BGP communities—and the decoder [explicitly asserts that it does not synthesize `community_id`](https://github.com/tenzir/mono/blob/d08047adc5ee977cbcd35a5b6ba67d2c9a4ef10f/engine/libtenzir/test/detail/netflow.cpp#L676). Suricata and Zeek provide it as enrichment, so those mappings are not analogous. I left it out and documented in 68cc8e2 that Community ID belongs in a separate enrichment step. Is there a specific exporter or enterprise IE you had in mind?

File: netflow/operators/ocsf/events/network_activity.tql [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+   $event.source.post_napt_destination_transport_port? != null {
+  $event.ocsf.dst_endpoint.proxy_endpoint = {
+    ip: move $event.source.post_nat_destination_ipv4_address? else
+        move $event.source.post_nat_destination_ipv6_address?,
+    port: move $event.source.post_napt_destination_transport_port?,
+  }
+}
+
+$event.ocsf.connection_info = {
+  direction_id: 0,
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql:120]: I kept this as Unknown in 68cc8e2 because the two fields describe different frames of reference. IPFIX IE 61 is ingress/egress at the Observation Point; OCSF `direction_id` is inbound/outbound/lateral relative to the monitored network boundary. Without network-scope context, mapping 0→Inbound and 1→Outbound would conflate those semantics. The raw `flow_direction` remains under `unmapped`. Is there context from `read_netflow` that would make the direct mapping safe?

File: netflow/operators/ocsf/events/network_activity.tql [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                       move $event.source.flow_end_seconds? else
+                       move $event.source.flow_end_milliseconds? else
+                       move $event.source.flow_end_microseconds? else
+                       move $event.source.flow_end_nanoseconds?
+
+$event.ocsf.time = $event.ocsf.start_time? else
+                   $event.ocsf.end_time? else
+                   $event.source.netflow.export_time?
+if $event.ocsf.start_time? != null and $event.ocsf.end_time? != null and
+   $event.ocsf.end_time >= $event.ocsf.start_time {
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql]: Fixed in 68cc8e2. Relative uptime fields now derive duration without header uptime and derive absolute start/end times when NetFlow v9 header uptime and export time are available, including the 32-bit uptime-counter wrap.

File: netflow/operators/ocsf/events/network_activity.tql [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  $event.ocsf.duration = count_milliseconds(
+    $event.ocsf.end_time - $event.ocsf.start_time,
+  ).round()
+}
+
+// --- OCSF: primary attributes -----------------
+
+$event.ocsf.src_endpoint = {
+  ip: move $event.source.source_ipv4_address? else
+      move $event.source.source_ipv6_address?,
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql]: Fixed in 68cc8e2. Source, destination, and post-NAT address selection now prefers the IPv6 IE when `ip_version == 6`; unselected address-family IEs remain under `unmapped`. Added coverage with populated IPv4 placeholders and real IPv6 addresses.

File: netflow/operators/ocsf/events/network_activity.tql [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  port: move $event.source.source_transport_port?,
+  mac: move $event.source.source_mac_address?,
+  interface_uid: (move $event.source.ingress_interface?).string(),
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql]: Fixed in 68cc8e2. `ingress_interface` and `egress_interface` now remain under `unmapped` rather than being attached to the remote endpoints.

File: netflow/operators/ocsf/events/network_activity.tql [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  },
+}
+
+$event.ocsf.connection_info = {
+  direction_id: 0,
+  protocol_num: move $event.source.protocol_identifier?,
+  protocol_ver_id: (move $event.source.ip_version?).int() else
+                   6 if ($event.ocsf.src_endpoint.ip?.is_v6() else false) or
+                        ($event.ocsf.dst_endpoint.ip?.is_v6() else false) else
+                   4,
```
[the remark this answers is not in the document]
Comment (mavam) [reply netflow/operators/ocsf/events/network_activity.tql]: Fixed in 68cc8e2. Missing `ip_version` is now inferred only from an endpoint IP; records without IP evidence emit Unknown (`protocol_ver_id: 0`). Added a layer-2 regression case.
